app/src/main/java/org/jellyfin/mobile/app/AppModule.kt

```diff
@@ -102,7 +107,21 @@ val applicationModule = module {
             }
         }
 
-        DefaultDataSource.Factory(context, baseDataSourceFactory)
+        val dataSourceFactory = DefaultDataSource.Factory(context, baseDataSourceFactory)
+
+        // Add authorization header. This is needed as we don't pass the
+        // access token in the url for Android Auto.
+        ResolvingDataSource.Factory(dataSourceFactory) { dataSpec: DataSpec ->
+            val authorizationHeaderString = AuthorizationHeaderBuilder.buildHeader(
+                clientName = apiClient.clientInfo.name,
+                clientVersion = apiClient.clientInfo.version,
+                deviceId = apiClient.deviceInfo.id,
+                deviceName = apiClient.deviceInfo.name,
+                accessToken = apiClient.accessToken,
+            )
+
+            dataSpec.withRequestHeaders(hashMapOf("Authorization" to authorizationHeaderString))
+        }
```

Comment:
Some sources, like Live TV and strm files do not load video through the Jellyfin server but from other origins directly. We should **never** sent the Jellyfin access token to those origins.